### ISSUE TITLE
Ugly source code colorscheme after change theme to "System theme"

### ISSUE THREAD

**Issue by moabson** [eclipse-platform/eclipse.platform.ui#176]
I'm using dark theme on gnome-shell and others tools. Some time ago I've been trying using dark theme in Eclipse making changes manually on "prefs" files in my workspace. But I was still not satisfied and tried to use "System theme" and Eclipse looks better for me, but I've some issues (Eclipse 2022-06):

1. After I've changed the theme to "System theme", source code looks by default like "Light" theme.

![image](https://user-images.githubusercontent.com/2575708/176345999-39177366-b92b-424b-b1db-e88ca2b1586f.png)

2. After I tried to customize colorscheme to one that I was doing I could not spot the properties to make the appropriate changes for "hover" over tabs and highlight of some piece of code that are still with grey color on bg.

![image](https://user-images.githubusercontent.com/2575708/176348384-ea93e02f-8f21-4366-bfb9-c58c7600e021.png)

![image](https://user-images.githubusercontent.com/2575708/176348273-cee1a2f6-56e1-4815-b8f8-8fa66db4ce3e.png)

Could someone assist me?

Best regards,
Moabson

**Comment by akurtakov** [comment]
System theme has been removed via https://github.com/eclipse-platform/eclipse.platform.ui/commit/9328b2d43cea202deb96f1ce88d73dd39acaa99d . If you want dark theme use the Dark theme as it's the only one that will adjust syntax colorschemes for dark background.

**Comment by mickaelistria** [comment]
Or you can try to disable theming so most info would be derived from your OS theme.
For the default Java syntax highlighting, I suggest you open a bug to JDT. It's the editor that should be responsible of dynamically picking a theme according to the background color. Some other plugins (eg TM4E) do it well, JDT could mimic that.

**Comment by moabson** [comment]
Thanks @mickaelistria, unchecking "Enable theming" the tabs looks better. I also found some properties for text highlight that can be helpful for others peoples with same issue:

Window -> Preferences -> General -> Editors -> Text Editors -> Annotations

I've changed all types that contains "Occurrence" keyword and was with grey color.

![Captura de tela de 2022-06-29 13-20-24](https://user-images.githubusercontent.com/2575708/176487955-f02da278-3efa-4d57-be02-dd2dfa1794c6.png)

By this time, Eclipse looks better (Linux) this way rather directly using the "Dark" theme option.
